### PR TITLE
Add form to app

### PR DESCRIPTION
**Pull request by fobadara**
The following requirements were met in this milestone:

- Create forms to perform the following functions:
    - Creates a Post on behalf of the current_user.
    - Create a comment on behalf of the current_user.

- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @comment = @post.comments.new
+  end
```
Comment (Lameck1) [inline app/controllers/comments_controller.rb]: - [ ] The work of the new action is only to make object instances available for the view that is going to use it. In this case, a view with some form for inputs.

  All the work done in this `new` action is kind of not necessary. We don't need to find the user in the `new` action or the post.
  
  We only need to ensure that a new instance of a Comment is available for the view that's going to help us create a new comment. We, therefore, need to clean up this `new` action. And other new actions in other controllers too.
  
  Consider this 👇🏼:
  
  ```ruby
  def new
    @comment = Comment.new
  end
  ```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -9,4 +9,20 @@ def show
...
+    @user = User.find(params[:user_id])
+    @post = @user.posts.create(post_params)
+    @post.save
+    redirect_to user_posts_path(@user)
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new
+  end
```
Comment (Lameck1) [inline app/controllers/posts_controller.rb]: - [ ] Let's get done with the `new` actions first. Please refactor this one too, just as discussed above.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @comment = @post.comments.new(text: comment_params[:text], post_id: @post.id, author_id: @user.id)
+
+    if @comment.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
```
Comment (Lameck1) [inline app/controllers/comments_controller.rb]: - [ ] Next, let's talk about the `create` actions. There is nothing wrong with this particular `create` action. We need to find the user and the specific post we are creating a comment for. However, in this particular task, there is this requirement quoted below:

  > - Create a comment on behalf of the current_user.
  
  This restricts us to only creating a comment for the current_user only. I hope you remember the current_user method in the Application Controller, we are supposed to utilize that here. In your current implementation, any user can create a comment. Please consider refactoring this as below:
  
 ```ruby
  def create
    # no need for finding the user for now, just do this:
    @comment = current_user.comments.new(comment_params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to user_post_path(current_user, @comment.post)
    else
      render :new
    end
  end
```

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class LikesController < ApplicationController
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @like = @user.likes.new(post_id: @post.id, author_id: @user.id)
+
+    redirect_to user_post_path(@user, @post) if @like.save
+  end
```
Comment (Lameck1) [inline app/controllers/likes_controller.rb:8]: No problem here. Any user can add a `like` 👍🏼.

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -9,4 +9,20 @@ def show
     @post = Post.find(params[:id])
     @comments = @post.comments
   end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.create(post_params)
+    @post.save
+    redirect_to user_posts_path(@user)
+  end
```
Comment (Lameck1) [inline app/controllers/posts_controller.rb]: - [ ] Let us refactor this one too as discussed above. Please utilize the `current_user` method in the Application Controller, to ensure we only create a `post` for the current user.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @fobadara, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlights
- Professional README ✔️
- Good commit messages ✔️ 

### Recommendation
- Please consider going through this [official rails guide](https://guides.rubyonrails.org/action_controller_overview.html#what-does-a-controller-do-questionmark) on the action controller.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, remember to tag me in your question so I can receive the notification.**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (fobadara) [reply app/controllers/comments_controller.rb]: Thank you

Comment (Lameck1) [reply app/controllers/comments_controller.rb]: You are welcome @fobadara.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Fobadara, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**